aws/resource_aws_sagemaker_model_test.go

```diff
@@ -0,0 +1,577 @@
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).sagemakerconn
+
+	req := &sagemaker.ListModelsInput{
+		NameContains: aws.String("terraform-testacc-sagemaker-model"),
+	}
+	resp, err := conn.ListModels(req)
+	if err != nil {
+		return fmt.Errorf("Error listing models: %s", err)
+	}
+
+	if len(resp.Models) == 0 {
+		log.Print("[DEBUG] No sagemaker models to sweep")
+		return nil
+	}
+
+	for _, model := range resp.Models {
+		_, err := conn.DeleteModel(&sagemaker.DeleteModelInput{
+			ModelName: model.ModelName,
+		})
+		if err != nil {
+			return fmt.Errorf(
+				"Error deleting sagemaker model (%s): %s",
+				*model.ModelName, err)
+		}
+	}
+
+	return nil
+}
+
+func TestAccAWSSagemakerModel_importBasic(t *testing.T) {
+	resourceName := "aws_sagemaker_model.foo"
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerModelConfig(rName),
+			},
+
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerModel_basic(t *testing.T) {
+	var model sagemaker.DescribeModelOutput
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerModelConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo", &model),
+
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_model.foo", "name", "terraform-testacc-sagemaker-model-foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "primary_container.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_model.foo", "primary_container.0.image",
+						"174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1"),
+					resource.TestCheckResourceAttrSet("aws_sagemaker_model.foo", "execution_role_arn"),
+					resource.TestCheckResourceAttrSet("aws_sagemaker_model.foo", "arn"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerModel_tags(t *testing.T) {
+	var model sagemaker.DescribeModelOutput
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerModelConfigTags(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo", &model),
+					testAccCheckSagemakerTags(&model, "foo", "bar"),
```

Comment:
This type of checks can be removed, too, correct? As I am checking against the state below.

Reply:
This makes sense to me!